symengine/dict.cpp

```diff
@@ -182,5 +204,22 @@ int map_uint_mpz_compare(const map_uint_mpz &A, const map_uint_mpz &B)
     return 0;
 }
 
+#if defined(HAVE_SYMENGINE_PIRANHA)
+int hash_set_compare(const hash_set &A, const hash_set &B)
+{
+    if (A.size() != B.size())
+        return (A.size() < B.size()) ? -1 : 1;
+    auto a = A.begin();
+    auto b = B.begin();
+    for (; a != A.end(); ++a, ++b) {
+        if ((*a).first != (*b).first)
+            return ((*a).first < (*b).first) ? -1 : 1;
+        if ((*a).second != (*b).second)
+            return ((*a).second < (*b).second) ? -1 : 1;
+    }
+    return 0;
+}
```

Comment:
This seems similar to `map_compare`, but it is optimized for integers only, since you use `<` instead of `__cmp__`. I think that's fine.